### ISSUE TITLE
Seems like memory leak

### ISSUE THREAD

**Issue by EvgenyGrishnov** [rebus-org/Rebus.RabbitMq#91]
Good day. Can't figure out where exactly this happens. 

**How to reproduce**

Create some .NET5 console app with BackgroundService

`        protected override Task ExecuteAsync(CancellationToken stoppingToken)
        {
            _bus = new BuiltinHandlerActivator();
            _bus .Handle<SomeMessage>(_apiNotifier.Process);

            Configure.With(_bus)
                .Logging(l => l.None())
                .Transport(t =>
                {
                    var builder = t.UseRabbitMq("amqp://localhost:5672/localdev", "MyQueue");
                    builder.Prefetch(10);
                })
                .Options(o =>
                {
                    o.SetNumberOfWorkers(2);
                    o.SetMaxParallelism(10);
                })
                .Start();

            return Task.CompletedTask;
        }`
Then just run debug and watch diagnostic tools "Process Memory" or JetBrains dotMemory profile.

**Comment by mookid8000** [comment]
Made a small program that just ran idle and let it run for a good hour or two - resulted in the following heap snapshot:
![image](https://user-images.githubusercontent.com/32968/145826802-ac872d49-3a23-4204-8fc3-a656149301c2.png)

Unless I am reading this wrong, it looks like looks of cancellation-related things are being left on the heap.

If anyone happens to know what can cause a heap snapshot to look like this, feel free to chime in 🙂

**Comment by mookid8000** [comment]
I've pushed Rebus.RabbitMq 7.3.4 to NuGet.org now. In it, I've made it block on receive, as it seemed from the memory snapshot above that loads and loads of OperationCancelledExceptions were not being collected - hopefully, the new, blocking way of receiving will not result in memory rising that that.

I've been running the same program as the one above with the debugger attached for almost 30 minutes now, and memory usage hasn't risen above 27 MB. I'll leave it running until tomorrow morning just to see if it works as it should.

Thanks for reporting this issue 🙂
